apps/studio/src/components/Datatable/Datatable.tsx

```diff
@@ -129,7 +129,7 @@ export const Datatable = <T extends object>({
         </Table>
       </Box>
       <Flex py="1rem" gap="1rem">
-        {pagination && (
+        {pagination && !!totalRowCount && totalRowCount > 0 && (
```

Comment:
i think the first check is sufficient

```suggestion
        {pagination && !!totalRowCount && (
```